### ISSUE TITLE
salt.cloud.clouds.vmware.revert_to_snapshot crash when using the default value

### ISSUE THREAD

**Issue by Shad0w1nk** [saltstack/salt#29488]
When using a command like: `salt-cloud -a revert_to_snapshot foo -y`

The following crash occur with salt 2015.8.1:
```
> There was an error actioning machines: 'NoneType' object has no attribute 'get'
> Traceback (most recent call last):
>   File "[...]/salt/cloud/cli.py", line 244, in run
>     ret = mapper.do_action(names, kwargs)
>   File "[...]/salt/cloud/__init__.py", line 1462, in do_action
>     vm_name, call='action'
>   File "[...]/salt/cloud/clouds/vmware.py", line 3277, in revert_to_snapshot
>     suppress_power_on = _str_to_bool(kwargs.get('power_off', False))
> AttributeError: 'NoneType' object has no attribute 'get'
```
Looking at the code in master (https://github.com/saltstack/salt/blob/develop/salt/cloud/clouds/vmware.py#L3172) the problem is still present. The argument kwargs should be check to make sure it's a dict.

I'm guessing a check like https://github.com/saltstack/salt/blob/develop/salt/cloud/clouds/vmware.py#L3101 should be done. I could provide a patch however this may take longer then needed. Still what would be the best step forward? Is there a test I should add to show the unexpected behavior or a strait fix?

**Comment by jfindlay** [comment]
@Shad0w1nk, thanks for the report.  You are welcome to send in a pull request to fix this.

**Comment by Shad0w1nk** [comment]
Is there any test I should write to exercise this bug? I'm not too familiar with salt and it's qa elements.

**Comment by jfindlay** [comment]
@Shad0w1nk, I'm not sure what might be most easily done here.  Do you have any ideas, @jtand?

**Comment by rallytime** [comment]
@Shad0w1nk Thanks for letting us know! You're right - we need to make sure kwargs is a dictionary first if we're going to perform a `kwargs.get`. I've fixed this with #29646 in the 2015.5 branch. It will be merged forward to the other branches within a couple of days.

**Comment by Shad0w1nk** [comment]
Awesome, thanks for doing the work!
